eodatasets3/wagl.py

```diff
@@ -121,6 +135,161 @@ def _unpack_products(
                     )
 
 
+def missing_packets_xml(granule):
+    """
+    Return children xml document if missing packets and None if not
+    """
+    l1_zip_path = Path(granule.wagl_metadata["source_datasets"]["source_level1"])
+    xml_file_path = Path(
+        granule.source_level1_metadata.accessories["metadata:s2_tile"].path
+    )
+
+    # Parse tile metadata xml file
+    with zipfile.ZipFile(l1_zip_path, "r") as z:
+        tree = Et.ElementTree(
+            Et.fromstring(z.read(xml_file_path.as_posix()), forbid_dtd=True)
+        )
+        root = tree.getroot()
+        msi_data_percentage = float(root[2][0][1].text)
+        if msi_data_percentage > 0.0:
+            children = root[2][1]
+            return children
+        else:
+            return None
+
+
+def mask_or_not(dataset: h5py.Dataset, granule):
+    """
+    Determine whether to mask h5 dataset
+    """
+    # Define bands to be masked
+    band_ids_mask = {str(i) for i in range(0, 13)}
+    band_ids_mask.add("8A")
+    band_bool = dataset.attrs["band_id"] in band_ids_mask
+    platform_bool = (
+        "sentinel" in granule.wagl_metadata["source_datasets"]["platform_id"].lower()
+    )
+
+    if platform_bool:
+        xml_bool = False if missing_packets_xml(granule) is None else True
+    else:
+        return False
+
+    # Mask only if all are True
+    return True if all([band_bool, platform_bool, xml_bool]) else False
+
+
+def mask_h5(dataset, granule, children):
```

Comment:
This is low priority, but I'm not a huge fan of passing "children" here as a few stray xml element objects. (this means every function needs knowledge of the xml file instead of just the function that parses it).